src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -144,9 +144,36 @@ class LibraryService (protected val argUserInfo: UserInfo, val rawlsDAO: RawlsDA
     }
   }
 
+  def updateAccess(docs: LibrarySearchResponse, workspaces: Seq[WorkspaceListResponse]) = {
+    val ids = workspaces collect {
+      case workspaceResponse: WorkspaceListResponse if (workspaceResponse.accessLevel != WorkspaceAccessLevels.NoAccess) =>
+        workspaceResponse.workspace.workspaceId
+    }
```

Comment:
open for discussion: is it better to have `workspaceAccess: true|false` as a Boolean in the response, or to copy the value of `workspaceResponse.accessLevel` into the response?

In the latter, we're returning much more granular info in case we want to use it in the UI - and it's more consistent with what the workspace listing returns.

Also, consider moving the `updateAccess` function into `LibraryServiceSupport` - rule of thumb here is that `LibraryService` contains the top-level methods (typically those called from the API routes), and `LibraryServiceSupport` contains utilities to support those top-level methods.